### PR TITLE
Fix email validator bug (uppercase domain)

### PR DESCRIPTION
**Pull request by shin10**
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] It's submitted to the `dev` branch for v1.x (or to a previous version branch), _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing: https://github.com/vuelidate/vuelidate/blob/master/.github/CONTRIBUTING.md#development-setup
- [x] New/updated tests are included

**Other information:**

The URL validator has been updated 3 years ago to accept mixed case domains:
https://github.com/vuelidate/vuelidate/issues/220

Somehow this has not been changed in the email validator, yet. This PR adds the ignore case modifier to the regex.

### CODE REVIEW

**Comment by dobromir-hristov** [comment]
before I merge this, do we have a concrete source, that emails should be case-insensitive? I thought that the part prior `@` is case-sensitive, and the domain after that isnt?

**Comment by shin10** [comment]
I found this:

> The local-part of a mailbox MUST BE treated as case sensitive.
   Therefore, SMTP implementations MUST take care to preserve the case
   of mailbox local-parts.  In particular, for some hosts, the user
   "smith" is different from the user "Smith".  However, exploiting the
   case sensitivity of mailbox local-parts impedes interoperability and
   is discouraged.  Mailbox domains follow normal DNS rules and are
   hence not case sensitive.
https://tools.ietf.org/html/rfc5321#section-2.4

In other words, you are right, especially for mail delivery.

For the validation of an input field however it means both (local-part and domain) can include lower and/or upper case letters.

**Comment by aethr** [comment]
@dobromir-hristov following the discussion in https://github.com/vuelidate/vuelidate/issues/753 I think it makes sense **not** to enforce lowercase on email domains in Vuelidate.

Because the domain is case-insensitive, `EXAMPLE.COM` and `Example.com` technically are both valid ways to express `example.com`. While I strongly feel that the domain part should be made lowercase *during validation against duplication* (and probably storage), that is probably a job best handled server-side and not in the front-end validation rules.

**Comment by shin10** [comment]
@dobromir-hristov 
Is there anything else, I can/should do?

**Review by dobromir-hristov — APPROVED** [review]

**Comment by bryce-mcmath** [comment]
@shin10 Do you know what release this change is/will be in?

**Comment by eduarguz** [comment]
@dobromir-hristov sorry to bother... May we have a release with this fix? I think this has not been tagged
Thanks!

**Comment by malinushj** [comment]
For those looking for an intermediary fix, here's my solution using a helper function:

```js
// force the domain of the email to be lowercase and send it to the email validator
export const emailFormatter = (value) => {
  if (!value) return value;
  let parts = value.split('@');
  parts[1] = parts[1].toLowerCase();
  return parts.join('@');
}
```

This is how it would be used:
```js
import { email, required } from 'vuelidate/lib/validators';
import { emailFormatter } from ...

// ...
validations: {
    form: {
      email: {
        required,
        email: (val) => email(emailFormatter(val)),
      },
    },
  },
// ...
```
